# CSE151A Group Project Milestone 2

In [8]:
!git clone git@github.com:gamroyan/cse151a-group-project.git

Cloning into 'cse151a-group-project'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.


## Data Exploration

In [7]:
from datasets import load_dataset
import pandas as pd

X = load_dataset("wykonos/movies")

print(X.shape)

print(X['train'].features.keys())

{'train': (722796, 20)}
dict_keys(['id', 'title', 'genres', 'original_language', 'overview', 'popularity', 'production_companies', 'release_date', 'budget', 'revenue', 'runtime', 'status', 'tagline', 'vote_average', 'vote_count', 'credits', 'keywords', 'poster_path', 'backdrop_path', 'recommendations'])


1. **How many observations does your dataset have?**

  - Our dataset has 722796 observations

2. **Describe all columns in your dataset their scales and data distributions. Describe the categorical and continuous variables in your dataset. Describe your target column and if you are using images plot some example classes of the images.**

  - Id - Unique ID for each movie
  - Title - Title of movie in observation
  - Genres - Genre of the movie in the observation
    - Categorical
  - Original language - The language that the movie is in
    - Categorical
  - Overview - Description of what the movie is about
  - Popularity - A scale of how popular a movie was based on votes and views
    - Continuous
    - Range from 0 - 6.68k
  - Production Companies - Company who made the film
  - Release Date - When the movie was released
  - Budget - Budget of the film in USD
    - Continuous
  - Revenue - How much money the film made
    - Continuous
  - Runtime - Length of the movie in minutes
    - Continuous
  - Status - Has the movie been released, cancelled, unreleased
    - Categorical
  - Tagline - Movie tagline
  - Vote average - Average public voting of the film
    - Scaled from 1-10
    - Continuous
  - Vote count - Hoe many votes were counted
    - Continuous
    - 0 - 33k
  - Credits - Actors in the movie
  - Keywords - Words used to categorize the movie
  - Poster path - link to poster
  - Backdrop path - link to backdrop

3. **Do you have missing and duplicate values in your dataset?**

  - Yes, there are missing values that have either a 0 or null in every category

**